src/components/JohnTheRipper/JohnTheRipper.tsx

```diff
@@ -0,0 +1,99 @@
+import { Button, LoadingOverlay, NativeSelect, NumberInput, Stack, TextInput, Title } from "@mantine/core";
+import { useForm } from "@mantine/form";
+import { useCallback, useState } from "react";
+import { CommandHelper } from "../../utils/CommandHelper";
+import ConsoleWrapper from "../ConsoleWrapper/ConsoleWrapper";
+import { writeTextFile, BaseDirectory } from "@tauri-apps/api/fs";
+
+interface FormValuesType {
+    filePath: string;
+    hash: string;
+    fileType: string;
+}
+
+const fileTypes = ["zip", "rar"];
+
+const JohnTheRipper = () => {
+    const [loading, setLoading] = useState(false);
+    const [output, setOutput] = useState("");
+    const [selectedFileTypeOption, setSelectedFileTypeOption] = useState("");
+
+    let form = useForm({
+        initialValues: {
+            filePath: "",
+            hash: "",
+            fileType: "",
+        },
+    });
+
+    const onSubmit = async (values: FormValuesType) => {
+        setLoading(true);
+
+        //if hash is not specified
+        if (values.hash.length === 0) {
+            const args = [`${values.filePath}`];
+
+            try {
+                const result = await CommandHelper.runCommand(`${values.fileType}2john`, args);
+                await writeTextFile("hash.txt", result, { dir: BaseDirectory.Temp });
+                setOutput(result);
+            } catch (e: any) {
+                console.log(e);
+                setOutput(e);
+            }
+
+            try {
+                const args = [`--format=${values.fileType}`, "/tmp/hash.txt"];
+                const result = await CommandHelper.runCommand("john", args);
+                console.log(result);
```

Comment:
Remove this (and others similar)

Reply:
This is required. Unless there is a way to shorten the code.
In the if true statement
The first command find the hash and stores it. The second reads the hash and cracks the password.
In the if else statement
Reads the user inputed hash and cracks the password.

Reply:
Sorry, meant the `console.log`

Reply:
Done